File: android/portal-sdk/src/main/java/com/amthalgroup/portal/PortalAuth.kt
package com.amthalgroup.portal

/**
 * Credentials handed to the portal over the bridge (`init.auth` / `auth`).
 * Credentials only ever travel in bridge messages — never in URLs.
 *
 * @param token backend bearer token, exactly as the portal's auth service expects.
 * @param branchId optional branch id the portal persists alongside the session.
 * @param userJson optional pre-built portal user object as a JSON string (the
 *   shape the portal persists under its `USER_KEY`). When omitted the portal
 *   reconstructs it from its profile endpoint using [token].
 */
data class PortalAuth(
    val token: String,
    val branchId: String? = null,
    val userJson: String? = null
) {
    /** Redact credentials from accidental logging. */
    override fun toString(): String =
        "PortalAuth(token=***, branchId=${if (branchId != null) "***" else "null"}, userJson=${if (userJson != null) "***" else "null"})"
}

File: ios/Sources/AmthalPortal/PortalAuth.swift
import Foundation

/// Credentials handed to the portal over the bridge (`init.auth` / `auth`).
/// Credentials only ever travel in bridge payloads — never in URLs.
public struct PortalAuth: Equatable, Sendable {
    /// Backend bearer token, exactly as the portal's auth service expects.
    public var token: String

    /// Optional branch identifier (`BRANCH_ID`).
    public var branchId: String?

    /// Optional raw JSON string of the pre-built portal user object (the shape
    /// the portal persists under `USER_KEY`). When omitted the portal
    /// reconstructs it by calling its profile endpoint with the token.
    public var userJSON: String?

    public init(token: String, branchId: String? = nil, userJSON: String? = nil) {
        self.token = token
        self.branchId = branchId
        self.userJSON = userJSON
    }
}
